overlord/devicestate/devicemgr.go

```diff
@@ -782,6 +782,20 @@ func (m *DeviceManager) doMarkSeeded(t *state.Task, _ *tomb.Tomb) error {
 	return nil
 }
 
+// canAutoRefresh is a helper that checks if the device is able to
+// auto-refresh
+func canAutoRefresh(st *state.State) bool {
+	// no need to wait for seeding on classic
+	if release.OnClassic {
+		return true
+	}
+
+	// on all-snap devices we need to be seeded first
+	var seeded bool
+	st.Get("seeded", &seeded)
+	return seeded
+}
```

Comment:
this looks like a good starting point, still wondering whether it will make sense to delay until device registration as well, though otoh we probably want to try to refresh (core in particular) if that fails for long